src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9086,11 +9086,29 @@ private ForStatementSyntax ParseForStatement(SyntaxList<AttributeListSyntax> att
                 // for a declaration first.
                 VariableDeclarationSyntax decl = null;
                 bool isDeclaration = false;
-                if (this.CurrentToken.Kind == SyntaxKind.RefKeyword)
+                bool haveScopedKeyword = false;
+
+                if (this.CurrentToken.ContextualKind == SyntaxKind.ScopedKeyword)
+                {
+                    if (this.PeekToken(1).Kind == SyntaxKind.RefKeyword)
+                    {
+                        isDeclaration = true;
+                    }
+                    else
+                    {
+                        this.EatToken();
+                        isDeclaration = ScanType() != ScanTypeFlags.NotType && this.CurrentToken.Kind == SyntaxKind.IdentifierToken;
```

Comment:
code below uses 'IsTrueIdentifier', any concern about this direct .IdentifierToken check?

Reply:
> code below uses 'IsTrueIdentifier', any concern about this direct .IdentifierToken check?

I don't think so. I think that, at most, that can affect how we parse error scenarios and I am fine to prioritize scoped type for now.